File: examples/javascript/quickstart.mjs
// Read a URL and search Hacker News — using the AgentSpan JS SDK.
//
// Prereq: agentspan serve --port 8080
// Run:    node examples/javascript/quickstart.mjs

import { AgentSpanClient } from "../../sdk/js/src/index.js";

const client = new AgentSpanClient({ baseUrl: "http://localhost:8080" });

// Read any URL — AgentSpan auto-detects the right channel.
const content = await client.read("https://news.ycombinator.com");
console.log(`Title: ${content.title || "(none)"}`);
console.log(`Body: ${content.body.slice(0, 200)}...`);
console.log(`Cached: ${content.cached}`);
console.log();

// Search Hacker News.
const results = await client.search("hackernews", "rust async", { limit: 5 });
console.log(`HN search results (${results.length}):`);
for (const r of results) {
  console.log(`  - ${r.title}`);
  console.log(`    ${r.url}`);
}

File: examples/rust/quickstart.rs
// AgentSpan Quickstart — Rust SDK
//
// Prereq: agentspan serve --port 8080
// Run:    cargo run --example quickstart

use agentspan_sdk::AgentSpanClient;

#[tokio::main]
async fn main() -> Result<(), Box<dyn std::error::Error>> {
    let client = AgentSpanClient::new("http://localhost:8080");

    // Read any URL.
    let content = client.read("https://example.com").await?;
    println!("Title: {:?}", content.title);
    println!("Body (first 200 chars): {}", &content.body[..200.min(content.body.len())]);
    println!("Cached: {}", content.cached);

    // Search Hacker News.
    let results = client.search("hackernews", "rust async", 10).await?;
    println!("\nHN search results ({}):", results.len());
    for r in &results {
        println!("  - {}", r.title);
        println!("    {}", r.url);
    }

    // List all channels.
    let channels = client.list_channels().await?;
    println!("\n{} channels available", channels.len());

    Ok(())
}
